src/TimePicker/TimePicker.js

```diff
@@ -122,20 +121,15 @@ const TimePicker = React.createClass({
     return {
       time: this._isControlled() ? this._getControlledTime() : this.props.defaultTime,
       dialogTime: new Date(),
-      muiTheme: this.context.muiTheme || getMuiTheme(),
     };
   },
 
-  componentWillReceiveProps(nextProps, nextContext) {
+  componentWillReceiveProps(nextProps) {
```

Comment:
This `newState = this.state` is unnecessary since React does the merging of state for you. It could be in fact dangerous in this case because we are mutating the state directly (`time`).